### PR TITLE
Added support for STM32G0 series

### PR DESCRIPTION
**Pull request by Jonathan-Hamberg**
Hi,
I have added support for the STM32G0 series chips listed on ST's website. [G0X1](https://www.st.com/content/st_com/en/products/microcontrollers-microprocessors/stm32-32-bit-arm-cortex-mcus/stm32-mainstream-mcus/stm32g0-series/stm32g0x1.html) [G0X0](https://www.st.com/content/st_com/en/products/microcontrollers-microprocessors/stm32-32-bit-arm-cortex-mcus/stm32-mainstream-mcus/stm32g0-series/stm32g0x0-value-line.html)

I tested the stm32-blinky and stm32-newlib applications with the STM32G071RB which is available as a nucleo development board.

I verified that all of the STM32G chips listed on ST's website compile without errors using the GCC Toolchain from ARM's website.

Let me know if there is anything that can be done to improve this pull request.

### CODE REVIEW

File: cmake/FindCMSIS.cmake [hunk 1]
```diff
@@ -96,9 +109,11 @@ ELSEIF(STM32_FAMILY STREQUAL "L1")
         SET(STM32Cube_DIR "/opt/STM32Cube_FW_L1_V1.8.0")
         MESSAGE(WARNING "No STM32Cube_DIR specified, using default: " ${STM32Cube_DIR})
     ENDIF()
+
```
Comment (Hish15) [inline cmake/FindCMSIS.cmake]: Those changes might be better of on another PR since they are not related to the support of the G0 serie

File: stm32-blinky/main.c [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
 #elif defined STM32F2
-# include <stm32f2xx_hal.h>
+#include <stm32f2xx_hal.h>
 #elif defined STM32F4
-# include <stm32f4xx_hal.h>
+#include <stm32f4xx_hal.h>
+#elif defined STM32G0
+#include <stm32g0xx_hal.h>
+#include <stm32g0xx_hal_gpio_ex.h>
+#include <stm32g0xx_hal_rcc.h>
```
Comment (Hish15) [inline stm32-blinky/main.c]: Aren't those useless since you include the stm32g0xx_hal.h file?

File: stm32-blinky/main.c [hunk 3]
```diff
@@ -1,9 +1,14 @@
 #if defined STM32F1
```
Comment (Hish15) [inline stm32-blinky/main.c]: Those changes are not related to the PR

File: stm32-blinky/main.c [hunk 4]
```diff
@@ -1,9 +1,14 @@
...
-# include <stm32f1xx_hal.h>
+#include <stm32f1xx_hal.h>
 #elif defined STM32F2
-# include <stm32f2xx_hal.h>
+#include <stm32f2xx_hal.h>
 #elif defined STM32F4
-# include <stm32f4xx_hal.h>
+#include <stm32f4xx_hal.h>
```
Comment (Hish15) [inline stm32-blinky/main.c]: Those changes are not related to the PR

**Comment by ObKo** [comment]
stm32-cmake rework in progress: https://github.com/ObKo/stm32-cmake/pull/117 can you check that G0 family support is implemented properly there?
